src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpHandlerResourceV3Provider.cs

```diff
@@ -28,16 +32,30 @@ public HttpHandlerResourceV3Provider()
 
             HttpHandlerResourceV3 curResource = null;
 
-            var clientHandler = CreateCredentialHandler(source.PackageSource);
+            if (source.PackageSource.IsHttp)
+            {
+                var clientHandler = CreateCredentialHandler(source.PackageSource);
 
-            // replace the handler with the proxy aware handler
-            curResource = DataClient.CreateHandler(clientHandler);
+                // replace the handler with the proxy aware handler
+                curResource = CreateHandler(source.PackageSource);
+            }
 
             return Task.FromResult(new Tuple<bool, INuGetResource>(curResource != null, curResource));
         }
 
+        private static HttpHandlerResourceV3 CreateHandler(PackageSource packageSource)
+        {
+            var handler = CreateCredentialHandler(packageSource);
+
+            var retryHandler = new RetryHandler(handler, maxTries: 3);
+
+            var resource = new HttpHandlerResourceV3(handler, retryHandler);
```

Comment:
The message handler is set up with source specific credentials, it's not the same between sources.